Cannot compile with cargo on Linux Arm (aarch64)

Hi Seth,

I just tried building perbase with cargo on a Linux aarch64 machine, and it's failing on rust-htslib:

```
rust$ cargo install perbase
    Updating crates.io index
  Downloaded perbase v0.10.1
  Downloaded 1 crate (77.2 KB) in 0.41s
  Installing perbase v0.10.1
    Updating crates.io index
     Locking 214 packages to latest compatible versions
      Adding bio v0.41.0 (available: v2.0.3)
      Adding env_logger v0.9.3 (available: v0.11.6)
      Adding gzp v0.10.1 (available: v0.11.3)
      Adding itertools v0.10.5 (available: v0.14.0)
      Adding rust-htslib v0.38.2 (available: v0.49.0)
  Downloaded env_logger v0.9.3
  Downloaded grep-cli v0.1.11
...
   Compiling rust-htslib v0.38.2
error[E0308]: mismatched types
   --> /lustre/software/rust/ampereone/1.84.0/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rust-htslib-0.38.2/src/bam/record.rs:152:16
    |
152 |             s: sam_copy.as_ptr() as *mut i8,
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `*mut u8`, found `*mut i8`
    |
    = note: expected raw pointer `*mut u8`
               found raw pointer `*mut i8`
```

I think this can be fixed by simply bumping the version of rust-htslib from 0.38 to 0.39 in Cargo.toml (v0.39 [introduced aarch64 support](https://github.com/rust-bio/rust-htslib/releases/tag/rust-htslib-v0.39.3)).

Would you accept a PR to do this?

Thanks,
Dave

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -39,7 +39,7 @@ lru_time_cache = "0.11.1"
 num_cpus = "1.13.0"
 rayon = "1.4.0"
 #TODO add  features = ["libdeflate"] when https://github.com/rust-bio/rust-htslib/pull/341 is merged
-rust-htslib = {version = "0.38"}
+rust-htslib = {version = "0.39"}
 rust-lapper = "1.0"
 serde = { version = "1.0.116", features = ["derive"] }
 smartstring = { version = "1.0.1", features = ["serde"] }

